utils/report_polling.py

```diff
@@ -0,0 +1,53 @@
+import logging
+import sys
+import threading
+import time
+from http import HTTPStatus
+from utils.logger import Logger
+
+
+class ReportsConstants:
+    UFM_API_REPORTS = 'reports'
+
+
+class ReportPolling(object):
+    def __init__(self, ufm_rest_client):
+        self.ufm_rest_client = ufm_rest_client
+        self.action_inprogress = False
+
+    def start_polling(self, report_id):
+        try:
+            self.action_inprogress = True
+            t = self.create_loading_thread()
+            report_status = None
+            while report_status != HTTPStatus.OK:
+                time.sleep(3)
+                report_response = self.ufm_rest_client.send_request(f"{ReportsConstants.UFM_API_REPORTS}/{report_id}")
+                if report_response.raise_for_status():
+                    break
+                report_status = report_response.status_code
+            self.action_inprogress = False
+            # move to new line after to avoid prining load icon and the message in the same line
+            print(f" ", end='\n')
+            print(report_response.json())
+            Logger.log_message(f"Report {report_id}: {report_response.json()}")
```

Comment:
I prefer to print the result in a file as well
let's add a new dir inside each report module and call it reports_result
the JSON response should be named with the creation timestamp